dns_test.go

```diff
@@ -374,17 +374,47 @@ func TestDNSProtocol(t *testing.T) {
 
 		_, port, _ := net.SplitHostPort(addr.String())
 
-		// Force IPv4
+		// Default DNS Server port
 		module := Module{
 			Timeout: time.Second,
 			DNS: DNSProbe{
 				QueryName: "example.com",
-				Protocol:  protocol + "4",
+				Protocol:  protocol,
 			},
 		}
 		recorder := httptest.NewRecorder()
 		registry := prometheus.NewRegistry()
-		result := probeDNS(net.JoinHostPort("localhost", port), recorder, module, registry)
+		result := probeDNS("localhost", recorder, module, registry)
+		if result {
+			t.Fatalf("DNS protocol: \"%v\" with localhost:53 succeeded, expected failure.", protocol)
```

Comment:
> if there is a local resolver localhost:53 on the test machine, but that's almost not.

A local resolver is very common, the machine I'm working on has one for example.

> Is there another better way to test it?

I can't think of one.